dashboard/lib/version_redirect_overrider.rb

```diff
@@ -0,0 +1,37 @@
+module VersionRedirectOverrider
```

Comment:
It would be nice to have a file- or module-level comment here giving a prose description of the desired redirect logic, and/or a link out to a spec describing the desired behavior in detail.  Even with the good tests you've written, I worry that the desired behavior is complex enough, and the related logic spread out enough, that it'll be challenging to reconstruct a mental model for these redirects in the future.

Reply:
yes, good call! i'll add a description and some usage info to the module